### PR TITLE
remove from string on indexuid and change indexuid internal representation

### LINKED ISSUE #3943: Remove the `From<String>` and `From<&str>` impl for IndexUid
It is a terrible footgun.
We used it extremely implictly in many places and it is extremely dangerous.

It assumes that the string is well-formed (with a ":") within.
We use it in countless places, especially in unit tests.

### PR DESCRIPTION
**Pull request by trinity-1686a**
### Description

fix #3943 
also fix *some* of the tests broken on main

### How was this PR tested?
unit tests

### CODE REVIEW

File: quickwit/quickwit-proto/src/getters.rs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+                self.$field
+                    .as_ref()
+                    .expect(concat!("`", stringify!($field), "` should be a required field in `", stringify!($struct), "`"))
+            }
+        }
+        )*
+    }
+}
+
+generate_getters! {
```
Comment (trinity-1686a) [inline quickwit/quickwit-proto/src/getters.rs:25]: we could call this macro multiple times to generate other getters for must-be-present Options

File: quickwit/quickwit-proto/protos/quickwit/common.proto [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+syntax = "proto3";
+
+package quickwit.common;
+
```
Comment (guilload) [inline quickwit/quickwit-proto/protos/quickwit/common.proto:23]: Please add a comment stating that the message is provided as extern path mapping to `crate::types::IndexUid`. See an example for `Position`.

File: quickwit/quickwit-proto/src/types/mod.rs [hunk 3]
```diff
@@ -158,42 +252,11 @@ pub struct InvalidIndexUid {
     pub invalid_index_uid_str: String,
 }
 
-impl From<&str> for IndexUid {
-    fn from(index_uid: &str) -> Self {
-        IndexUid::from(index_uid.to_string())
-    }
-}
+impl TryFrom<String> for IndexUid {
```
[the remark this answers is not in the document]
Comment (guilload) [reply quickwit/quickwit-proto/src/types/mod.rs]: `feature = "postgres" then

[the remark this answers is not in the document]
Comment (guilload) [reply quickwit/quickwit-proto/src/types/mod.rs]: + comment

**Review by guilload — APPROVED** [review]

**Pull request merged**